### PR TITLE
JSONrpc Public api tests

### PR DESCRIPTION
**Pull request by modship**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

image

### CODE REVIEW

File: massa-api/src/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//! Copyright (c) 2023 MASSA LABS <info@massa.net>
+//!
+//!
+mod apiv2;
+mod mock;
+mod public;
+// mod server;
```
Comment (aoudiamoncef) [inline massa-api/src/tests/mod.rs:7]: could be  deleted

File: massa-api/src/tests/public.rs [hunk 2]
```diff
@@ -0,0 +1,1167 @@
...
+};
+use massa_protocol_exports::{
+    test_exports::tools::{create_block, create_endorsement, create_operation_with_expire_period},
+    MockProtocolController,
+};
+use massa_signature::KeyPair;
+use massa_time::MassaTime;
+use serde_json::Value;
+use tempfile::NamedTempFile;
+// use serde_json::Value;
```
Comment (aoudiamoncef) [inline massa-api/src/tests/public.rs:53]: could be removed

**Review by aoudiamoncef — APPROVED** [review]
LTGM. Well done

**Review by aoudiamoncef — APPROVED** [review]

**Comment by modship** [comment]
bors merge

**Pull request merged**